plugins/woocommerce/client/blocks/assets/js/blocks/product-collection/edit/inspector-controls/taxonomy-controls/index.tsx

```diff
@@ -36,35 +40,61 @@ export const useTaxonomies = (): Taxonomy[] => {
 	}, [ taxonomies ] );
 };
 
+/**
+ * Normalize the name so first letter of every word is capitalized.
+ */
+const normalizeName = ( name: string | undefined | null ) => {
+	if ( ! name ) {
+		return '';
+	}
+
+	return name
+		.split( ' ' )
+		.map( ( word ) => word.charAt( 0 ).toUpperCase() + word.slice( 1 ) )
+		.join( ' ' );
+};
+
 function TaxonomyControls( {
 	setQueryAttribute,
 	trackInteraction,
 	query,
-}: QueryControlProps ) {
+	collection,
+}: QueryControlProps & { collection: string | undefined } ) {
 	const { taxQuery } = query;
 
 	const taxonomies = useTaxonomies();
-	if ( ! taxonomies || taxonomies.length === 0 ) {
-		return null;
-	}
 
-	/**
-	 * Normalize the name so first letter of every word is capitalized.
-	 */
-	const normalizeName = ( name: string | undefined | null ) => {
-		if ( ! name ) {
-			return '';
+	const filteredTaxonomies = useMemo( () => {
+		if ( ! taxonomies || taxonomies.length === 0 ) {
+			return [];
+		}
+
+		// For filters panel, show the complementary taxonomy.
+		if ( collection === CoreCollectionNames.BY_CATEGORY ) {
+			return taxonomies.filter(
+				( taxonomy ) => taxonomy.slug === 'product_tag'
+			);
+		}
+		if ( collection === CoreCollectionNames.BY_TAG ) {
+			return taxonomies.filter(
+				( taxonomy ) => taxonomy.slug === 'product_cat'
+			);
```

Comment:
This logic has to be reverted from filtering to excluding.

Tags and Categories are taxonomies added by WooCommerce Core but there may be more taxonomies like brands etc. and they're all handled generically in here. That means, this would ignore other potential taxonomies and just left one. Should be something like: 

```suggestion
		// For filters panel, show the complementary taxonomy.
		if ( collection === CoreCollectionNames.BY_CATEGORY ) {
			return taxonomies.filter(
				( taxonomy ) => taxonomy.slug !== 'product_cat'
			);
		}
		if ( collection === CoreCollectionNames.BY_TAG ) {
			return taxonomies.filter(
				( taxonomy ) => taxonomy.slug !== 'product_tag'
			);
```

Reply:
Thanks, I have addressed it.